api/namex/services/name_request/abstract_name_request.py

```diff
@@ -99,12 +113,15 @@ def next_state_code(self, val):
 
     @classmethod
     def set_request_type(cls, entity_type, request_action):
```

Comment:
generally this is not supplied by the front-end and will set by the event listener on the request model except in the case of an XPRO/Assumed name.

Reply:
Really not sure why you have this mapping.  I think it should be removed.  The request_type if it is supplied should just be included as any other field in the request attributes update (header) other wise the event listener is setting it.  please take this out.

Reply:
I don't think we're on the same page. These properties are required when mapping data, and are not exclusive to the PATCH. They are also used for mapping PUT and POST data. Doesn't work without those changes in there, as the ENTITY uses request_action_cd but the frontend uses request_action. However the frontend does something weird which is returning the entity with request_action as the property instead of using request_action_cd. Like if the pay load looks like: 
```
{ request_action: 'blah', data: { <entity fields> }
but instead i get:
{ data: { <entity fields>, request_action: 'blah', request_action_cd: 'blah' } where request_action_cd is the actual value of the entity. Don't pass weird fields in the entity from the frontend. 

Anyway, without a change on the frontend, I need to check for the existence of either code. Not having this change breaks stuff.
```

Reply:
We will remove the event listener for def update_request_action_entity_type in the request model. and use this set_request_type(cls, entity_type, request_action):  However, if the payload supplies the requestTypeCd we just set it and dont use the mapper.